pkg/relayer/miner/miner.go

```diff
@@ -88,18 +89,30 @@ func (mnr *miner) mapMineRelay(
 ) (_ either.Either[*relayer.MinedRelay], skip bool) {
 	relayBz, err := relay.Marshal()
 	if err != nil {
+		meteringError := mnr.relayMeter.SetNonApplicableRelayReward(ctx, relay.GetReq().GetMeta())
+		if meteringError != nil {
+			return either.Error[*relayer.MinedRelay](fmt.Errorf("failed to unclaim relay price: %w", meteringError)), false
+		}
 		return either.Error[*relayer.MinedRelay](err), false
 	}
 	relayHashArr := protocol.GetRelayHashFromBytes(relayBz)
 	relayHash := relayHashArr[:]
 
 	relayDifficultyTargetHash, err := mnr.getServiceRelayDifficultyTargetHash(ctx, relay.Req)
 	if err != nil {
+		meteringError := mnr.relayMeter.SetNonApplicableRelayReward(ctx, relay.GetReq().GetMeta())
+		if meteringError != nil {
+			return either.Error[*relayer.MinedRelay](fmt.Errorf("failed to unclaim relay price: %w", meteringError)), false
+		}
 		return either.Error[*relayer.MinedRelay](err), false
 	}
 
 	// The relay IS NOT volume / reward applicable
 	if !protocol.IsRelayVolumeApplicable(relayHash, relayDifficultyTargetHash) {
+		meteringError := mnr.relayMeter.SetNonApplicableRelayReward(ctx, relay.GetReq().GetMeta())
+		if meteringError != nil {
+			return either.Error[*relayer.MinedRelay](fmt.Errorf("failed to unclaim relay price: %w", meteringError)), false
+		}
```

Comment:
Wdyt about:

1. extracting this to another method to de-duplicate (readability/maintainability)
2. starting a pkg/relayer/miner/errors.go and:
    -  promoting this error to a sentinel cosmos-sdk error (consistency)
    - pushing the error construction down into `#SetNonApplicableRelayReward()` so we only need to `return either.Error[*relayer.MinedRelay](err)` here